### PR TITLE
Update verify release set

### PR DESCRIPTION
**Pull request by JimSuplizio**
**Currently, verify release set works at follows:**
Given the service directory and a list of libraries to be released verify that the list of libraries has full transitive closure. This means that we have to look at each library's pom file's current dependencies and ensure that each dependency is contained within the
list to be released.

**The update I'm making here is:**
If the _current_ dependency isn't on the list of libraries to be released, instead of failing, check Maven to see if it was already released and, if so, allow the release to proceed. The scenario here would be releasing a library for a patch that's using the released versions of libraries within the same service directory. In other words, the _current_ dependency on another library in the same service directory has already been released. This would also help if someone accidentally unchecked a checkbox for an was able to release everything else in the service directory and had to start another run to release the single library.

It's worth noting that the IsMavenPackageVersionPublished has been being used for several years to check whether or not something exists before creating the release/tag and it's tried and true, I'm just using it.

### CODE REVIEW

File: eng/versioning/verify_release_set.ps1 [hunk 1]
```diff
@@ -8,6 +8,7 @@ param(
 )
 
 . "${PSScriptRoot}/../common/scripts/common.ps1"
+. "${PSScriptRoot}/../scripts/Language-Settings.ps1"
```
Comment (weshaggard) [inline eng/versioning/verify_release_set.ps1:11]: Language settings is already pulled in by common so this shouldn't be needed.

File: eng/versioning/verify_release_set.ps1 [hunk 2]
```diff
@@ -81,11 +84,25 @@ if ($ArtifactsList -and $ArtifactsList.Count -gt 0) {
...
+                                $releasedLibToVerify = "$($libraryToVerify):$($version)"
+                                Write-Host "$releasedLibToVerify is a current dependency of $($inputGroupId):$($inputArtifactId) and is not on the release list, checking if $version has been published."
+                                if (!$mavenPublished.ContainsKey($releasedLibToVerify)) {
+                                    $released = IsMavenPackageVersionPublished -pkgId $artifactId -pkgVersion $version -groupId $groupId
+                                    $mavenPublished.Add($releasedLibToVerify, $released)
+                                }
+                                # If the current dependency isn't on the command line to be released and the version
+                                # isn't published on Maven, then report an error
+                                if (!$mavenPublished[$releasedLibToVerify]) {
+                                    if (!$missingLibraries.ContainsKey($libraryToVerify) -and !$mavenPublished) {
```
Comment (weshaggard) [inline eng/versioning/verify_release_set.ps1:98]:
```suggestion
                                    if (!$missingLibraries.ContainsKey($libraryToVerify)) {
```

Not sure why we needed that mavenPublished check as it always is non-null.

**Review by weshaggard — APPROVED** [review]
Couple minor comments but otherwise looks good.

File: eng/versioning/verify_release_set.ps1 (hunk 1 above)
Comment (JimSuplizio) [reply eng/versioning/verify_release_set.ps1]: Good catch, removed with second commit.

File: eng/versioning/verify_release_set.ps1 (hunk 2 above)
Comment (JimSuplizio) [reply eng/versioning/verify_release_set.ps1]: Also gone. That was going to be something different before I'd simplified it.

**Comment by JimSuplizio** [comment]
There's nothing in any CI pipeline that'll test this. I've run a [template release](https://dev.azure.com/azure-sdk/internal/_build/results?buildId=1680890&view=results) against this refs/pull to verify.

**Pull request merged**